javalin/src/main/java/io/javalin/http/HttpCode.kt

```diff
@@ -63,13 +63,28 @@ enum class HttpCode(val status: Int, val message: String) {
     LOOP_DETECTED(508, "Loop Detected"),
     @Deprecated("Obsoleted by https://datatracker.ietf.org/doc/status-change-http-experiments-to-historic/")
     NOT_EXTENDED(510, "Not Extended"),
-    NETWORK_AUTHENTICATION_REQUIRED(511, "Network Authentication Required");
+    NETWORK_AUTHENTICATION_REQUIRED(511, "Network Authentication Required"),
+    CUSTOM_HTTP_CODE(599, "Non-standard HTTP code");
 
     override fun toString(): String =
         "$status $message"
 
     companion object {
-        fun forStatus(status: Int) = values().find { it.status == status }
+
+        /** Returns the HttpCode for the given status code. Works for all status codes between 100 and 599. */
+        fun forStatus(status: Int): HttpCode? {
+
+            if (status !in 100..599) return null //Values between 100 and 599 are valid HTTP codes.
+
+            val find = values().find { it.status == status }
+
+            if (find == null) {
+                CUSTOM_HTTP_CODE.status = status; //Allow for custom codes
+                return CUSTOM_HTTP_CODE
+            }
```

Comment:
1. I think we should cache `values()` in static field in general for performance reasons
2. I don't think this custom status code is needed here at all, especially that this impl effectively works only for one value. I'd leave it as it was before tbh.